src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -1,34 +1,32 @@
 package akka.persistence.cassandra.journal
 
-import java.lang.{ Long => JLong }
+import java.lang.{Long => JLong}
 import java.nio.ByteBuffer
 
-import com.datastax.driver.core.policies.{LoggingRetryPolicy, RetryPolicy}
-import com.datastax.driver.core.policies.RetryPolicy.RetryDecision
-
-import scala.concurrent._
-import scala.collection.immutable.Seq
-import scala.collection.JavaConversions._
-import scala.math.min
-import scala.util.{Success, Failure, Try}
-
-import akka.persistence.journal.AsyncWriteJournal
 import akka.persistence._
 import akka.persistence.cassandra._
+import akka.persistence.journal.AsyncWriteJournal
 import akka.serialization.SerializationExtension
-
 import com.datastax.driver.core._
+import com.datastax.driver.core.policies.RetryPolicy.RetryDecision
+import com.datastax.driver.core.policies.{LoggingRetryPolicy, RetryPolicy}
 import com.datastax.driver.core.utils.Bytes
+import com.typesafe.config.Config
+
+import scala.collection.immutable.Seq
+import scala.concurrent._
+import scala.math.min
+import scala.util.{Failure, Success, Try}
 
-class CassandraJournal extends AsyncWriteJournal with CassandraRecovery with CassandraConfigChecker with CassandraStatements {
+class CassandraJournal(cfg: Config) extends AsyncWriteJournal with CassandraRecovery with CassandraConfigChecker with CassandraStatements {
 
-  val config = new CassandraJournalConfig(context.system.settings.config.getConfig("cassandra-journal"))
+  val config = new CassandraJournalConfig(cfg)
   val serialization = SerializationExtension(context.system)
 
   import config._
 
-  val cluster = clusterBuilder.build
-  val session = cluster.connect()
+  lazy val cluster = clusterBuilder.build
+  lazy val session = cluster.connect()
```

Comment:
I was too brief in my request for this. All the session related things should be lazy. It is used in the constructor below, e.g. `val preparedWriteMessage`. I think it is easies to break out all those things into a class

```
class CassandraSession {
  val cluster = clusterBuilder.build
  val session = cluster.connect()
  if (config.keyspaceAutoCreate) {
     retry(config.keyspaceAutoCreateRetries) {
       session.execute(createKeyspace)
     }
   }
   session.execute(createTable)
   session.execute(createMetatdataTable)
   session.execute(createConfigTable)
   val preparedWriteMessage = session.prepare(writeMessage)
   ...
}
```

and 

```
lazy val session = new CassandraSession
```